python/cudf/cudf/core/_internals/where.py

```diff
@@ -234,9 +238,19 @@ def where(
 
     if isinstance(frame, DataFrame):
         if hasattr(cond, "__cuda_array_interface__"):
-            cond = DataFrame(
-                cond, columns=frame._column_names, index=frame.index
-            )
+            if (
+                isinstance(cond, DataFrame)
+                or isinstance(cond, cupy._core.core.ndarray)
+                or isinstance(cond, cuda.cudadrv.devicearray.DeviceNDArray)
```

Comment:
Do we need to generalize this to include other array-like objects? Alternately, is the `else` case on line 249 possibly more tightly scoped?

Reply:
I checked for these objects because they were causing some cases for `test_df_sr_mask_where()` to fail - I'm not sure whether there are other cases to check for